src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -153,6 +162,26 @@ protected void tick(@Nonnull Block b) {
                         nextFluid.setType(Material.AIR);
                     }
 
+                    return;
+                }
+                if (SlimefunUtils.isItemSimilar(menu.getItemInSlot(slot), emptyBottle, true, false)) {
+                    ItemStack bottle = getFilledBottle(fluid);
+
+                    if (!menu.fits(bottle, getOutputSlots())) {
+                        return;
+                    }
+
+                    Block nextFluid = findNextFluid(fluid);
+
+                    if (nextFluid != null) {
+                        removeCharge(b.getLocation(), ENERGY_CONSUMPTION);
+                        menu.consumeItem(slot);
+                        menu.pushItem(bottle, getOutputSlots());
+                        if(ThreadLocalRandom.current().nextInt(100) < 30) {
+                            nextFluid.setType(Material.AIR);
+                        }
```

Comment:
Formatting, also... why make it randomized
```suggestion
                        menu.pushItem(bottle, getOutputSlots());

                        if (ThreadLocalRandom.current().nextInt(100) < 30) {
                            nextFluid.setType(Material.AIR);
                        }
```

Reply:
https://github.com/Slimefun/Slimefun4/pull/3451#discussion_r795150129

This was my reasoning though as stated something to discuss rather than just add.

Reply:
If I were to respect vanilla mechanics then I would just add an random chance since water bottles doesn't consume water off from an water source but then sefi stated it does on cauldron, kinda middle in there but I like the fact sefi stated this as in favor of staying in vanilla